packages/interactivity/src/vdom.ts

```diff
@@ -100,7 +100,8 @@ export function toVdom( root: Node ): Array< ComponentChild > {
 					const namespace = regexResult?.[ 1 ] ?? null;
 					let value: any = regexResult?.[ 2 ] ?? attributeValue;
 					try {
-						value = value && JSON.parse( value );
+						const parsedValue = JSON.parse( value );
+						value = isObject( parsedValue ) ? parsedValue : value;
 					} catch ( e ) {}
```

Comment:
Nit: we can omit `( e )` when not using the error:
```suggestion
					} catch {}
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/61960/commits/b7b6da8f447a974daf807c8034119ccd6824fcaa